MaraBot/Core/CommandUtils.cs

```diff
@@ -84,5 +86,70 @@ public static async Task<string[]> MentionRoleWithoutPing(CommandContext ctx, Di
             var hasPerms = await HasBotPermissions(ctx, Permissions.MentionEveryone);
             return roles.Select(r => r.IsMentionable || hasPerms ? $"@({r.Name})" : r.Mention).ToArray();
         }
+
+        /// <summary>
+        /// Grants a role to a user.
+        /// </summary>
+        public static async Task GrantRoleAsync(CommandContext ctx, string roleString)
+        {
+            // Grant user their new role.
+            var newRole = ctx.Guild.Roles
+                .FirstOrDefault(kvp => roleString.Equals(kvp.Value.Name));
+
+            if (newRole.Value == null)
+            {
+                await ctx.RespondAsync(
+                    "No role set for access to spoiler channel.\n" +
+                    "This shouldn't happen! Please contact your friendly neighbourhood developers!");
```

Comment:
I understand the context of this error message, but the function name suggests a more generic error, like
```c#
$"Role {newRole} has not been found in guild {ctx.Guild.Name}."
```
This also allows to use an `errorMessage` variable, like you do with the other functions.

Reply:
Fixed in 61553b5ec999f733c98aafec29f9b806df76d04d.